src/hooks/useTabVisibility.ts

```diff
@@ -0,0 +1,37 @@
+import { useEffect, useState } from 'react'
+
+const useTabVisibility = () => {
+  const [isTabVisible, setIsTabVisible] = useState(true)
+
+  useEffect(() => {
+    const onFocus = () => {
+      onVisibilityChange()
+    }
+
+    const onVisibilityChange = () => {
```

Comment:
I put cooldown period of 60sec to publish visibility state from true to false, to not make so much request if user switches tab or window frequently.